src/otr.h

```diff
@@ -54,14 +57,12 @@ void otr_log (IRC_CTX *server, const char *to,
 /* own */
 
 #include "config.h"
-
 #include "otr-formats.h"
 
-/* 
- * maybe this should be configurable?
- * I believe bitlbee has something >500.
- */
-#define OTR_MAX_MSG_SIZE 400
+/* Most IRC servers limit messages to 512 bytes in length, 
+ * including trailing whitespace and the CR-LF terminating characters 
+ * i.e  len( bytes(msg) | <0x20> x ... | <0x0D0A> ) <= 512 bytes */
+#define OTR_MAX_MSG_SIZE 510 //RFC 2812
```

Comment:
I'm not certain this makes sense. The message size often needs to account for the `:user!user@hostname ...` bits at the start. I think hexchat internally accounts that for like 90ish characters by default. You can kinda figure it out exactly if you know the information ahead of time.

Reply:
Yeah, I'm not totally sure myself. I relied on this [](https://modern.ircdocs.horse/#message-format) and [RFC2812](https://datatracker.ietf.org/doc/html/rfc2812.html#section-2.3) in coming up with that number. IRC seems to have a limit of 512, but hexchat could certainly modify this.

Reply:
The RFC is referring to the entire line. But generally at the layer of users like this message just means the content.

So an IRC line is 512 bytes but with protocol overhead user messages can only realistically use 400-500 depending on the users full host.

Reply:
Yeah, I can confirm. I just tried sending a line of 561 chars (including spaces) on Libera.chat, and hexchat split it off at 446 chars.

Reply:
If you open `Window > Raw Log` you can see why it had to be split with the prefixes.